app/services/middleware_topology_service.rb

```diff
@@ -33,14 +36,28 @@ def entity_display_type(entity)
 
   def build_entity_data(entity)
     data = build_base_entity_data(entity)
-    data.merge!(:status => 'Unknown',
-                :display_kind => entity_display_type(entity))
-    data[:icon] = entity.decorate.try(:listicon_image)
+    data[:status] = 'Unknown'
+    data[:display_kind] = entity_display_type(entity)
+
+    # fonticon takes precedence before the listicon_image
+    # and both methods on decorator takes precedence before the global icons hash from the UiServiceMixin
+    data[:icon] = if !(icon = entity.decorate.try(:fonticon)).nil?
```

Comment:
You are correct. I am starting to think that putting font icons in decorators is not a good idea: 
it doesn't contain font family there, it has no utf char there that is needed here, etc.
and there is again duplication - decorator and uimixin both have the same information that needs to be maintained twice. plus, the icons are still repeated in every entity in topology json, hence increasing the json size.
@pilhuhn I'm adding you to this discussion, I sense we need some "cleaner" approach for font icons and would like to hear your ideas.